### PR TITLE
Add data stream docs to Fleet user guide

### PR DESCRIPTION
**Pull request by bmorelli25**
### Summary

This PR adds a general overview of data streams to the Fleet user guide. The content is not complete yet, but I don't have any additional time to devote to this in the current iteration. I'd like to publish this and circle back later with additional updates.

Unbeknownst to DeDe and I, it appears that the Elasticsearch documentation team has already put a lot of work into creating docs about data streams–as expected, some of this content is particularly useful for Fleet/Agent users. I'm not going to reinvent the wheel, so this PR borrows very heavily from the following sources:

* https://www.elastic.co/guide/en/elasticsearch/reference/7.11/data-streams.html
* https://www.elastic.co/blog/an-introduction-to-the-elastic-data-stream-naming-scheme

Eventually, we may want to single-source some snippets of content from the ES repository.

### Documentation preview

**[Click here to view a documentation preview](https://observability-docs_429.docs-preview.app.elstc.co/guide/en/fleet/master/data-streams.html)**

### Related issues

For https://github.com/elastic/apm-server/issues/4494.

### CODE REVIEW

File: docs/en/ingest-management/data-streams.asciidoc [hunk 1]
```diff
@@ -0,0 +1,105 @@
+[[data-streams]]
+= Data streams
+
+{agent} uses data streams to easily store append-only time series data across multiple indices
+while giving you a single named resource for requests.
```
Comment (dedemorton) [inline docs/en/ingest-management/data-streams.asciidoc:5]:
```suggestion
{agent} uses data streams to store time series data across multiple indices
while giving you a single named resource for requests.
```

Comment (dedemorton) [reply docs/en/ingest-management/data-streams.asciidoc:5]: My brain tripped over this sentence a couple of times. It's maybe too packed.  Is the term append-only needed?

File: docs/en/ingest-management/data-streams.asciidoc [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+Data streams are well-suited for logs, metrics, traces, and other continuously generated data.
+They offer a host of benefits over other indexing strategies:
+
+* *Reduced number of fields per index*: Indices only need to store a specific subset of your
+data–meaning no more indices with hundreds of thousands of fields.
+This leads to better space efficiency and faster queries.
+As an added bonus, only relevant fields are shown in Discover.
+
+* *More granular data control*: For example, filesystem, load, cpu, network, and process metrics are sent
+to different indices–each potentially with its own rollover, retention, and security permissions.
```
Comment (dedemorton) [inline docs/en/ingest-management/data-streams.asciidoc:15]:
```suggestion
to different indices–-each potentially with its own rollover, retention, and security permissions.
```

Comment (dedemorton) [reply docs/en/ingest-management/data-streams.asciidoc:15]: I think -- gets converted into an em dash.

File: docs/en/ingest-management/data-streams.asciidoc [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+* *Flexible*: Use the custom namespace component to divide and organize data in a way that
+makes sense to your use-case or company.
```
Comment (dedemorton) [inline docs/en/ingest-management/data-streams.asciidoc:18]:
```suggestion
makes sense to your use case or company.
```

File: docs/en/ingest-management/data-streams.asciidoc [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+The `dataset` is defined by the integration and describes the ingested data and its structure for each index.
+For example, you might have a dataset for process metrics with a field describing whether the process is running or not,
+and another dataset for disk I/O metrics with a field describing the number of bytes read.
++
+For APM data, the `dataset` is the instrumented service's `service.name`.
+
+`namespace`::
+A user-configurable arbitrary grouping, such as an environment (`dev`, `prod`, or `qa`),
+a team, or a strategic business unit.
+A `namespace` can be up to 100 bytes in length (multi-byte characters will count toward this limit faster).
```
Comment (dedemorton) [inline docs/en/ingest-management/data-streams.asciidoc:43]:
```suggestion
A `namespace` can be up to 100 bytes in length (multibyte characters will count toward this limit faster).
```

Comment (dedemorton) [reply docs/en/ingest-management/data-streams.asciidoc:43]: Both forms are correct, but I think all one word is in the spirit of Elastic style.

File: docs/en/ingest-management/data-streams.asciidoc [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+// Corresponds to the `data_stream.dataset` field.
+
+The naming scheme separates each components with a `-` character:
+
+[source,text]
+--
+<type>-<dataset>-<namespace>
+--
+
+For example, if you've set up the Nginx integration with a namespace of `prod`,
```
Comment (dedemorton) [inline docs/en/ingest-management/data-streams.asciidoc:54]: You might want to format this differently to make it easier to scan. Maybe more white space? Honestly I think the line wraps in the middle of names are making it harder to follow the example.

File: docs/en/ingest-management/data-streams.asciidoc [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+
+All data streams, and the pre-built dashboards that they ship with,
+are viewable on the Fleet Data Streams page:
+
+[role="screenshot"]
+image::images/kibana-fleet-datastreams.png[Data streams page]
+
+TIP: If you're familiar with the concept of indices, you can think of each data stream as a separate index in {es}.
+Under the hood though, things are a bit more complex.
+All of the juicy details are available in {ref}/data-streams.html[{es} Data streams].
```
Comment (dedemorton) [inline docs/en/ingest-management/data-streams.asciidoc:68]: Nice! Knowing "juicy details" is way more intriguing than having to read some other docs. :->

**Review by dedemorton — COMMENTED** [review]
Nice work! Thanks for doing this.

I have a few minor suggestions (mostly related to dashes...that's how you know the work is good...haha).

File: docs/en/ingest-management/data-streams.asciidoc (hunk 1 above)
Comment (bmorelli25) [reply docs/en/ingest-management/data-streams.asciidoc:5]: You're right; we don't need that level of detail here.

File: docs/en/ingest-management/data-streams.asciidoc (hunk 2 above)
Comment (bmorelli25) [reply docs/en/ingest-management/data-streams.asciidoc:15]: That's already an em dash :)

Comment (bmorelli25) [reply docs/en/ingest-management/data-streams.asciidoc:15]: It's hard to tell in the source file, but a little easier in the built version:
Screen Shot 2021-03-10 at 7 17 13 PM

File: docs/en/ingest-management/data-streams.asciidoc (hunk 5 above)
Comment (bmorelli25) [reply docs/en/ingest-management/data-streams.asciidoc:54]: I see what you mean.

Comment (bmorelli25) [reply docs/en/ingest-management/data-streams.asciidoc:54]: I added additional white space and moved the examples to code blocks in 2ddd984.

**Pull request merged**